### PR TITLE
Update docker environment and CONTRIBUTING.md instructions

### LINKED ISSUE #55: CONTRIBUTING.md documentation is outdated
If I follow the steps listed in CONTRIBUTING.md file, I get the following error when I execute `dev/bin/php composer install`:

<details>
 <summary>Your requirements could not be resolved to an installable set of packages.</summary>

```
$ dev/bin/php composer install
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/
Loading composer repositories with package information
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "4.4.*"
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested package symfony/framework-bundle ^5.3 exists as symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these are rejected by your constraint.
  Problem 2
    - The requested package symfony/security-bundle ^5.3 exists as symfony/security-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.23, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these are rejected by your constraint.
  Problem 3
    - The requested package symfony/browser-kit ^5.3 exists as symfony/browser-kit[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.22, v4.4.24, v4.4.25, v4.4.27, v4.4.3, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these are rejected by your constraint.
  Problem 4
    - doctrine/doctrine-bundle 2.4.3 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0|^6.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.4.2 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0|^6.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.4.1 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0|^6.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.4.0 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0|^6.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.3.2 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.3.1 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.3.0 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.2.4 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.2.3 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.2.2 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.2.1 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.2.0 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.1.2 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.1.1 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.1.0 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.9 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.8 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.7 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.6 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.5 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.4 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.3 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.2 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.10 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.1 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - doctrine/doctrine-bundle 2.0.0 requires symfony/framework-bundle ^3.4.30|^4.3.3|^5.0 -> satisfiable by symfony/framework-bundle[4.4.x-dev, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.10, v4.4.11, v4.4.12, v4.4.13, v4.4.14, v4.4.15, v4.4.16, v4.4.17, v4.4.18, v4.4.19, v4.4.2, v4.4.20, v4.4.21, v4.4.22, v4.4.24, v4.4.25, v4.4.26, v4.4.27, v4.4.3, v4.4.30, v4.4.31, v4.4.4, v4.4.5, v4.4.6, v4.4.7, v4.4.8, v4.4.9] but these conflict with your requirements or minimum-stability.
    - Installation request for doctrine/doctrine-bundle ^2.0 -> satisfiable by doctrine/doctrine-bundle[2.0.0, 2.0.1, 2.0.10, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.0.9, 2.1.0, 2.1.1, 2.1.2, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.3.0, 2.3.1, 2.3.2, 2.4.0, 2.4.1, 2.4.2, 2.4.3].
```

</details>

On the other hand, the development tools listed are a bit outdated:

  - COMPOSER_VERSION=1.9.3
  - FLEX_VERSION=1.6.2
  - PHP_CS_FIXER_VERSION=2.16.2
  - XDEBUG_VERSION=2.9.2

So my question is, do you think we must update the Dockerfile to use a more up-to-date environment, or drop the whole `dev` directory and rewrite the CONTRIBUTING.md file?

### PR DESCRIPTION
**Pull request by ajgarlag**
Alternative to #56 updating current docker environment.

Fix #55

### CODE REVIEW

**Comment by chalasr** [comment]
@ajgarlag I'm divided. I would drop the docker setup, but since you updated it here I guess we can keep it if some find it useful. What's your preference?

**Comment by ajgarlag** [comment]
@chalasr I've decided to submit two pull requests so you can evaluate the pros and cons of each.

When I contributed the [`authorization_code`](https://github.com/trikoder/oauth2-bundle/pull/18) grant to `trikoder/oauth2-bundle`, the docker environment was a great help mainly because the `timecop` extension was used for time-sensitive tests. Without the docker environment, maybe I would not have contributed the PR. But it has a main drawback: it requires docker as a development dependency (I know several PHP developers that are not comfortable with this tool).

In this project, I've not needed docker at all, and the contributing workflow was similar to other PHP projects. But @X-Coder264 already [commented](https://github.com/thephpleague/oauth2-server-bundle/issues/55#issuecomment-943161584) the advantages of keeping it.

The main problem I see with keeping the docker environment is that is not being used in GitHub actions. I think that if we keep the docker environment as the official workflow for contributing to the bundle, we should update GitHub actions to make use of it so we can ensure that the docker environment is working as expected.

**Comment by chalasr** [comment]
Oh, I missed Antonio's comment on your issue sorry. Let's keep it.

> The main problem I see with keeping the docker environment is that is not being used in GitHub actions. I think that if we keep the docker environment as the official workflow for contributing to the bundle, we should update GitHub actions to make use of it so we can ensure that the docker environment is working as expected.

Agree 👍

**Comment by ajgarlag** [comment]
I've added two GitHub actions to check CS and execute tests using the docker environment.

Does the same statement from Antonio Pauletich's (I'm Antonio too) comment about not having PHP-CS-Fixer as part of the dev dependencies applies to Psalm? Should we move Psalm from composer.json to docker environment?

**Comment by chalasr** [comment]
> (I'm Antonio too)

Haha right, let's use gh usernames then.

> Should we move Psalm from composer.json to docker environment?

That makes sense to me.

**Comment by ajgarlag** [comment]
Ok, one last question. I think we can execute `checking coding standards`, `running static analysis`, and `running unit tests` sequentially in one GitHub action to avoid building the docker environment three times. WDYT?

**Comment by chalasr** [comment]
I like the fact they represent distinct checks on a PR so better keep duplicating (or reuse steps somehow?) IMO

**Comment by ajgarlag** [comment]
In that case, IMO the PR is ready for me. If it's OK for you, I'll squash the commits.

File: .github/workflows/docker-coding-standards.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "checkout"
+        uses: "actions/checkout@v2"
+
+      - name: "build the environment"
+        run: "dev/bin/docker-compose build"
+
+      - name: "install dependencies"
+        run: "dev/bin/php composer install"
```
Comment (chalasr) [inline .github/workflows/docker-coding-standards.yml:17]: do we really need to install dependencies for this job?

Comment (ajgarlag) [reply .github/workflows/docker-coding-standards.yml:17]: Yes.

```
 dev/bin/php psalm --shepherd --stats
Could not find any composer autoloaders in /app/src/
Add a --root=[your/project/directory] flag to specify a particular project to run Psalm on.
```

 But is not needed for checking coding standards. I'll update the PR

**Comment by chalasr** [comment]
@ajgarlag The added `docker-*` workflows can replace the existing (non docker) ones isn't it? 
I don't think it's relevant to have the same job runs with and without docker. Let's just run everything on docker?

**Comment by ajgarlag** [comment]
> @ajgarlag The added `docker-*` workflows can replace the existing (non docker) ones isn't it? I don't think it's relevant to have the same job runs with and without docker. Let's just run everything on docker?

Yes, but in that case, the code will be tested only with the versions defined in Dockerfile. So my previous proposition was to keep current GitHub actions that will test the code with the latest version of each tool, and to add a new docker.yml file to run `checking coding standards`, `running static analysis`, and `running unit tests` sequentially just to ensure that the docker environment is working. If this action fails, but the other one passes, then the Dockerfile should be updated.

**Comment by X-Coder264** [comment]
@ajgarlag The Dockerfile could be changed so that it fetches the latest versions of Composer/Flex/Xdebug/PHP CS fixer by default and then the same Docker file could be used to test across all supported [PHP versions](https://github.com/trikoder/oauth2-bundle/blob/v3.2.0/.github/workflows/tests.yaml#L33).

**Comment by ajgarlag** [comment]
@X-Coder264 Thanks, I'll try to update de Dockerfile.

**Comment by ajgarlag** [comment]
I've updated the Dockerfile to install always the latest version of development tools. The only build argument for the docker image is the PHP version.

WDYT?

**Comment by ajgarlag** [comment]
I've updated the Dockerfile to install always the latest version of development tools. The only build argument for the docker image is the PHP version.

WDYT?

**Comment by chalasr** [comment]
@ajgarlag Looks great. One thing, can we reduce the number of checks? 
The coverage before this PR was good enough to me, it's ok not to test with low/high deps on all php versions

**Comment by ajgarlag** [comment]
@chalasr After some vacation days, I've rebased and finished my work here. WDYT?

**Review by chalasr — APPROVED** [review]
Awesome!

**Comment by ajgarlag** [comment]
I've changed the `8.1-rc` docker tag so the latest available version is used always

**Comment by chalasr** [comment]
Thank you @ajgarlag.

**Comment by ajgarlag** [comment]
@chalasr Off-topic: While coding this PR, I got a failed test that I cannot explain. See https://github.com/thephpleague/oauth2-server-bundle/runs/4177437425?check_suite_focus=true. Maybe a time sensitive assertion?

**Pull request merged**